src/core/utils/types/basic-details-type.ts

```diff
@@ -0,0 +1,11 @@
+import { Pid } from "./ids";
+
+export interface BasicDetailsType {
```

Comment:
If all the props can be optional (undefined) I think you should wrap the whole interface in `Partial`:
https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype

Reply:
I am not sure I get this.....

Reply:
Either like [this](https://www.typescriptlang.org/play?ssl=10&ssc=1&pln=11&pc=1#code/PTAECgFMA8AcHsBOAXUyCetKgGIEMBXAZ2QEkATUAXlAAMASAbwDsCBbAI0kQF9aBuKHCSoMWUAAUAlpRoMW7LrwC0TEoinMA5jwBcTfMTLk+g8GOwA5AgBsbeDjcgAeACoA+aqEagA2gGlQTVAAa0h0eAAzUFcAXV0YgNjQAB9QVjtQHjNNZG5IvABjbAAhPCIpQoARSGQ8KRsiV0xIMqJsRnBQUEJkAAskIgT1TS1BbtgZBOlycdAYPMRmPBsJRHhyAkLjYeQNbTm2PEWpFeasXf2xrtBySCJCjVhkKXhmS9G59Eg8RA+Dm4vZBOf7XHjgcwtUBtSo1OoNJpQmjWOwOJzOCS-F4rZww6q1eqNc6tcqQdzuQRAA) or like [this](https://www.typescriptlang.org/play?#code/PTAECgFMA8AcHsBOAXUyCetKgGIEMBXAZ2QEkATUAXlAAMASAbwDsCBbAI0kQF9aBuKHCSoMWUAAUAlpRoMW7LrwC0TEoinMA5jwBcTfMTLk+g8GOwB5WMinxmeADYBBZuQByBR47wdHkAB4AFQA+alBGUABtAGlQTVAAa0h0eAAzUCCAXQB+XUzYrNAAH1BWb1AeM01kbjS8AGNsACE8IikGgBFIZDwpRyIgzEhWomxGcFBQQmQACyQifPVNLUEp2Bl86XI10BhaxAdHCUR4cgIG4yXkDW1dtjwDqSchrGvb1cnQckgiBo0bHZmO8Vrt0JA8IgQXcvrZkP5oZ8eOBzMNQKMOt1ev1BmiaNZbPYnK4PF4fH5Ahiuj0+gNXiM2pAQoIgA)